src/sentry/static/sentry/app/views/acceptOrganizationInvite.tsx

```diff
@@ -0,0 +1,220 @@
+import React, {MouseEvent} from 'react';
+import styled from 'react-emotion';
+import {browserHistory} from 'react-router';
+
+import {logout} from 'app/actionCreators/account';
+import {t, tct} from 'app/locale';
+import {urlEncode} from '@sentry/utils';
+import Alert from 'app/components/alert';
+import AsyncView, {AsyncViewProps, AsyncViewState} from 'app/views/asyncView';
+import Button from 'app/components/button';
+import ConfigStore from 'app/stores/configStore';
+import Link from 'app/components/links/link';
+import NarrowLayout from 'app/components/narrowLayout';
+import SettingsPageHeader from 'app/views/settings/components/settingsPageHeader';
+import space from 'app/styles/space';
+
+type InviteDetails = {
+  orgSlug: string;
+  needsAuthentication: boolean;
+  needs2fa: boolean;
+  needsSso: boolean;
+  existingMember: boolean;
+  ssoProvider?: string;
+};
+
+type Props = AsyncViewProps;
+type State = AsyncViewState & {
+  inviteDetails: InviteDetails;
+  accepting: boolean | undefined;
+  acceptError: boolean | undefined;
+};
+
+class AcceptOrganizationInvite<P extends Props, S extends State> extends AsyncView<P, S> {
```

Comment:
@EvanPurkhiser Forgive me if this is a silly question, but why do we need to extend `AsyncViewProps` and `AsyncViewState` in generic types here? Are we using these generic types later in the class?

Reply:
Tbh i actually don't know _exactly_ what's going on here, I actually just copied from the `AsyncView`.

Reply:
@EvanPurkhiser LOL ok

Reply:
@dashed maybe can shed some light

Reply:
I think that's only required in AsyncView so that other components can extend from it, not relevant here

Reply:
@EvanPurkhiser If we don't need it here, I'd prefer removing it.